generators/client/templates/react/src/main/webapp/app/shared/util/pagination-utils.ts

```diff
@@ -0,0 +1,38 @@
+import { ITEMS_PER_PAGE } from '../../shared/util/pagination.constants';
+
+export interface IPaginationState {
+  itemsPerPage: number;
+  sort: string;
+  order: string;
+  activePage: number;
+}
+
+export const getUrlParameter = (name, search) => {
+  name = name.replace(/[\[]/, '\\[').replace(/[\]]/, '\\]');
+  const regex = new RegExp('[\\?&]' + name + '=([^&#]*)');
+  const results = regex.exec(search);
+  return results === null ? '' : decodeURIComponent(results[1].replace(/\+/g, ' '));
+};
+
+export const getSortState = location => {
+  const itemsPerPage = ITEMS_PER_PAGE;
+  const pageParam = getUrlParameter('page', location.search);
+  const sortParam = getUrlParameter('sort', location.search);
+  let sort = 'id';
+  let order = 'asc';
+  let activePage = 1;
+  if (pageParam !== '' && !isNaN(parseInt(pageParam, 10))) {
+    activePage = parseInt(pageParam, 10);
+  }
+  if (sortParam !== '') {
+    sort = sortParam.split(',')[0];
+    order = sortParam.split(',')[1];
+  }
+  return { itemsPerPage, sort, order, activePage };
+};
+
+export const getItemsNumber = (totalItems, itemsPerPage) => {
+  const division = Math.round(totalItems / itemsPerPage);
```

Comment:
I think it should be `Math.floor(totalItems / itemsPerPage)`.

For example with `totalItems = 35` and `itemsPerPage = 20`, division is 2 so the return is 2+1=3 (instead of 2).